app/cdap/components/common/DrawerWidget/index.tsx

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Container, Drawer, DrawerProps, IconButton, Typography } from '@material-ui/core';
+import grey from '@material-ui/core/colors/grey';
+import ChevronLeftRoundedIcon from '@material-ui/icons/ChevronLeftRounded';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import { Underline } from 'components/common/DrawerWidget/IconStore/Underline';
+import React, { ReactNode } from 'react';
+import styled from 'styled-components';
+
+export interface IDrawerWidgetProps extends DrawerProps {
+  headingText?: ReactNode | string;
+  showDivider?: boolean;
+  headerActionTemplate?: ReactNode | JSX.Element;
+  closeClickHandler?: () => void;
+  showBackIcon?: boolean;
+}
+
+const BackIcon = styled(ChevronLeftRoundedIcon)`
+  font-size: 40px;
+  color: ${grey[600]};
+`;
+
+const CustomizedIconButton = styled(IconButton)`
+  padding: 0px;
+`;
+
+const CloseIcon = styled(CloseRoundedIcon)`
+  font-size: 30px;
+  cursor: pointer;
+  margin: 5px;
+`;
+
+const CustomizedDrawer = styled(Drawer)`
+  & .MuiDrawer-paper {
+    top: 46px;
+    height: calc(100vh - 47px);
+    width: 500px;
+  }
+`;
+
+const Divider = styled.div`
+  width: 1px;
+  height: 28px;
+  background-color: #dadce0;
+  margin: 0 15px;
+`;
+
+const DrawerContainer = styled(Container)`
+  height: 100%;
+  padding: 0px 30px;
+`;
+
+const DrawerHeader = styled.header`
+  height: 60px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding-left: 15px;
+  padding-right: 18px;
+`;
+
+const HeaderActions = styled(Box)`
+  display: flex;
+  align-items: center;
+`;
+
+const Label = styled(Typography)`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 20px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: #000000; /* Mui Colors not available */
+`;
+
+const LabelContainer = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+`;
+
+export default function({
+  headingText,
+  open,
+  showDivider,
+  headerActionTemplate,
+  children,
+  closeClickHandler,
+  showBackIcon,
+  anchor = 'right',
+}: IDrawerWidgetProps) {
+  return (
+    <CustomizedDrawer anchor={anchor} open={open} data-testid="import-data-widget">
+      <DrawerHeader>
+        <HeaderActions>
+          {showBackIcon && (
+            <CustomizedIconButton aria-label="back-icon" data-testid="back-icon">
+              <BackIcon />
+            </CustomizedIconButton>
+          )}
+          <LabelContainer>
+            <Label data-testid="widget-heading-text" component="span">
+              {headingText}
+            </Label>
+            <Underline />
```

Comment:
Let's double-check, but I don't think we want an underline here

Reply:
@njbriggs above underline is displayed under the widget header title and it is needed.

Reply:
according to the design, we need to have this Nick. Please let us know if we need to discuss this further.
Thank you.

Reply:
Alright, but we've previously asked for this to be removed from the design.